Development/ssl/ssl_utils.cpp

```diff
@@ -0,0 +1,282 @@
+#include "ssl/ssl_utils.h"
+
+#include <boost/algorithm/string.hpp> // for boost::split
+#include <openssl/err.h>
+#include <openssl/pem.h>
+
+namespace ssl
+{
+    namespace experimental
+    {
+        namespace details
+        {
+            // get common name from subject
+            std::string common_name(X509* x509)
+            {
+                auto subject_name = X509_get_subject_name(x509);
+                if (!subject_name)
+                {
+                    throw ssl_exception("failed to get subject: X509_get_subject_name failure: " + last_openssl_error());
+                }
+                auto name = X509_NAME_oneline(subject_name, NULL, 0);
+                std::string subject(name);
+                OPENSSL_free(name);
+
+                // exmaple subject format
+                // e.g. subject=/DC=Example/CN=api.example.com
+                const std::string common_name_prefix{ "CN=" };
+                std::vector<std::string> tokens;
+                boost::split(tokens, subject, boost::is_any_of("/"));
+                auto found_common_name_token = std::find_if(tokens.begin(), tokens.end(), [&common_name_prefix](const std::string& token)
+                    {
+                        return std::string::npos != token.find(common_name_prefix);
```

Comment:
Shouldn't this check that the token _starts_ with `CN=`, not just _contains_ it?